src/test/java/io/lettuce/SslTest.java

```diff
@@ -36,40 +40,70 @@
 import io.lettuce.core.*;
 import io.lettuce.core.api.StatefulRedisConnection;
 import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.codec.StringCodec;
+import io.lettuce.core.masterslave.MasterSlave;
 import io.lettuce.core.pubsub.api.async.RedisPubSubAsyncCommands;
 import io.lettuce.core.pubsub.api.sync.RedisPubSubCommands;
 import io.netty.handler.codec.DecoderException;
 import io.netty.handler.ssl.OpenSsl;
 
+
 /**
  * @author Mark Paluch
  */
 public class SslTest extends AbstractTest {
 
     private static final String KEYSTORE = "work/keystore.jks";
     private static final String TRUSTSTORE = "work/truststore.jks";
-    private static RedisClient redisClient;
+    private static final File TRUSTSTORE_FILE = new File(TRUSTSTORE);
+    private static final int MASTER_SLAVE_BASE_PORT_OFFSET = 2000;
 
-    private static final RedisURI URI_NO_VERIFY = RedisURI.Builder.redis(host(), sslPort()) //
-            .withSsl(true) //
+    private static final RedisURI URI_NO_VERIFY = sslURIBuilder(0) //
             .withVerifyPeer(false) //
             .build();
 
-    private static final RedisURI URI_VERIFY = RedisURI.Builder.redis(host(), sslPort(1)) //
-            .withSsl(true) //
+    private static final RedisURI URI_VERIFY = sslURIBuilder(1) //
             .withVerifyPeer(true) //
             .build();
 
-    private static final RedisURI URI_CLIENT_CERT_AUTH = RedisURI.Builder.redis(host(), sslPort(2)) //
-            .withSsl(true) //
+
+    private static final RedisURI URI_CLIENT_CERT_AUTH = sslURIBuilder(2) //
             .withVerifyPeer(true) //
             .build();
 
+    private static final List<RedisURI> MASTER_SLAVE_URIS_NO_VERIFY = Arrays.asList(
```

Comment:
Let's keep the tests simple by not re-testing functionality that is already covered, in particular, we should just have tests for the regular test cases with and without verifications. Client cert/OpenSSL/various trust-stores test cases are already covered and don't require own tests.

Reply:
My first thought here was that there's enough subtlety that it deserves extra test coverage, but I guess at the end of the day, they all end up just creating one or more `StatefulRedisConnection`s, so I can get onboard w/ them being covered. I'll clean those up now.